### DISCUSSION TITLE
Proposal: Repeating action

### DISCUSSION THREAD

**Discussion by mosesnagar** [dotnet/csharplang#5585] [Language Ideas] [answered]
Hi all,

something you need repeat some code for X times. I mean code like that:
`
for(var i=0; i<X;i++) {
  ....
  var x = SomeGenerator();
  SomeAction(x);
  ......
}
`

I found some ways to do it online, for example:
`Enumerable.Repeat<Action>(() => { Console.WriteLine("Hello World"); this.DoSomething(); },10).ToList().ForEach(f => f.Invoke());`
`
Enumerable.Repeat<Action>(() => 
{
    Console.WriteLine("Hello World.");
    this.DoSomeStuff();
}, 10).ToList().ForEach(x => x());
`
and more as you can see here for example: 
https://stackoverflow.com/questions/6406741/is-there-an-elegant-way-to-repeat-an-action

but I don't find it very elegant. I thought about method extension for action, so we can do:
`SomeAction().Repeat(times)` or `Repeat(aciton, times) or Repeat(() => { ... }, times)`

WDYT?

**Comment by huoyaoyuan** [comment, marked as answer]
This does not require any language change. It's part of the library.

**Reply by mosesnagar** [reply]
you mean it's related to dotnet/runtime and not here?

**Reply by ufcpp** [reply]
yes, dotnet/runtime.

**Reply by mosesnagar** [reply]
thanks!

**Comment by TahirAhmadov** [comment]
I have this extension method: (feel free to use - my Christmas gift to the world)
```cs
public static class _Int32_Extensions
{
  public static void Repeat(this int This, Action action)
  {
    for(int i = 0; i < This; ++i) action();
  }
  public static async Task RepeatAsync(this int This, Func<Task> action)
  {
    for(int i = 0; i < This; ++i) await action();
  }
}
3.Repeat(()=>Console.Write("ho "));
```
